File: benchmarks/go_comparison/strings.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func record(index int64, active bool) string {
	var output strings.Builder
	output.Grow(64)
	output.WriteString("customer-")
	output.WriteString(strconv.FormatInt(index, 10))
	output.WriteString(":active=")
	output.WriteString(strconv.FormatBool(active))
	output.WriteString(":balance=")
	output.WriteString(strconv.FormatInt(index*3, 10))
	return output.String()
}

func main() {
	fmt.Println(record(0, true))

	var total int
	var index int64
	active := true
	for index < 300_000 {
		total += len(record(index, active))
		active = !active
		index++
	}
	fmt.Println(total)
}

File: benchmarks/rust_comparison/strings.rs
use std::fmt::Write;

fn record(index: i64, active: bool) -> String {
    let mut output = String::with_capacity(64);
    write!(
        &mut output,
        "customer-{index}:active={active}:balance={}",
        index * 3
    )
    .unwrap();
    output
}

fn main() {
    println!("{}", record(0, true));

    let mut total: usize = 0;
    let mut index: i64 = 0;
    let mut active = true;
    while index < 300_000 {
        total += record(index, active).len();
        active = !active;
        index += 1;
    }
    println!("{total}");
}
